wsit/ws-sx/wssx-impl/src/main/java/com/sun/xml/wss/impl/transform/AttachmentCompleteTransform.java

```diff
@@ -72,4 +80,9 @@ protected XMLSignatureInput enginePerformTransform(
    public boolean wantsNodeSet ()       { return true; }
    public boolean returnsOctetStream () { return true; }
    public boolean returnsNodeSet ()     { return false; }
+
+    // Required by Santuario 2.2.X
+   protected XMLSignatureInput enginePerformTransform(XMLSignatureInput input, OutputStream os, Element transformElement, String baseURI, boolean secureValidation) throws IOException, CanonicalizationException, InvalidCanonicalizerException, TransformationException, ParserConfigurationException, SAXException {
+       return null;
```

Comment:
it is always needed to nullify the result?

Reply:
See upstream

Reply:
This method is only required in Santuario 2.2.X which upstream doesn't use

Reply:
I'm not convinced this is correct.
Upstream uses Santuario 2.1.7, where `TransformSpi` has three `enginePerformTransform` methods available for overriding, of which they overwrite just the one implemented already above (leaving the other two parent methods to throw `UnsupportedOperationException`).

Santuario 2.2.3 has only one method for overriding, with which we override it to do nothing but return null.
The pre-existing `enginePerformTransform` method already implemented above no longer overrides anything and is just a standalone method now.